codersdk/richparameters.go

```diff
@@ -104,6 +97,41 @@ func validateBuildParameter(richParameter TemplateVersionParameter, buildParamet
 	return validation.Valid(richParameter.Type, current, previous)
 }
 
+// inOptionSet returns if the value given is in the set of options for a parameter.
+func inOptionSet(richParameter TemplateVersionParameter, value string) bool {
+	optionValues := make([]string, 0, len(richParameter.Options))
+	for _, option := range richParameter.Options {
+		optionValues = append(optionValues, option.Value)
+	}
+
+	// This is the simple case
+	if slice.Contains(optionValues, value) {
+		return true
+	}
+
+	// If the type is `list(string)` and the form_type is `multi-select`, then we check each individual
+	// value in the list against the option set. If the form_type is the empty string, we have to consider the
+	// possibility it might be a multi-select. Old provisioners do not send us the form_type.
+	mightBeMultiSelect := richParameter.Type == provider.OptionTypeListString && (richParameter.FormType == string(provider.ParameterFormTypeMultiSelect) || richParameter.FormType == "")
+	if !mightBeMultiSelect {
+		return false
+	}
+
+	var checks []string
+	err := json.Unmarshal([]byte(value), &checks)
+	if err != nil {
+		return false
+	}
+
+	for _, check := range checks {
+		if !slice.Contains(optionValues, check) {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
I'd like to have a unit test for this in `richparameters_internal_test.go`.

Reply:
Added